doc/HelpInterface.md

```diff
@@ -0,0 +1,89 @@
+# Halibot Help Text Interface
+
+Help text is crucial for new users to learn how to operate modules and Halibot itself.
+While Halibot core does not require module developers to supply help text, it is greatly encouraged to do so.
+This document will define the help text ecosystem that is strongly recommended to adhere to.
+
+# Sample Usage
+
+## How to get help text?
+
+### Via an agent (or "By chatting with it")
+
+Halibot will provide a "Help" module as a default that needs to be enabled.
+This handles the "!help" command, which has the following usage:
+
+```
+someuser> !help
+halibot1> Your friendly bot! Use !help <module> to get specific module help
+halibot1> Loaded modules: admin, help, quote
+
+someuser> !help quote
+halibot1> Display quotes from a file. Available commands are !quote, !quoteadd, !quotedel
+
+someuser> !help quote !quoteadd
+halibot1> Add a new quote to the db
+halibot1> Call like so: !quoteadd text of new quote here
+
+```
+
+### Via command line
+
+*do we actually want to support this?*
```

Comment:
I suspect would could do something similar here where we just call the class' help text, but that does weird things when there's runtime changes. We may not actually want this.